tests/release/pipelines/push_disk_images_to_cdn.go

```diff
@@ -0,0 +1,276 @@
+								}
+							}
+						}
+					}
+					return nil
+				}, releasecommon.ReleaseCreationTimeout, releasecommon.DefaultInterval).Should(Succeed())
+			})
+
+		})
+	})
+})
+
+func createRHAIEnterpriseContractPolicy(rhaiECPName string, managedFw framework.Framework, devNamespace, managedNamespace string) {
+	releasePublicKeyDecoded := []byte("-----BEGIN PUBLIC KEY-----\n" +
+                                        "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEHHTvfqOgdrdt9TXDyYDMYlwZ8r8r\n" +
+                                        "AsiNjiVlB/DrkEdj3mNgxW7Papn1MgtxqRL/gJ3Jql2Z9tOYKNA08ojFAg==\n" +
+                                        "-----END PUBLIC KEY-----")
+	Expect(managedFw.AsKubeAdmin.TektonController.CreateOrUpdateSigningSecret(
+			releasePublicKeyDecoded, releasecommon.PublicSecretNameAuth, managedNamespace)).To(Succeed())
+	err = managedFw.AsKubeAdmin.CommonController.LinkSecretToServiceAccount(managedNamespace, releasecommon.PublicSecretNameAuth, rhaiServiceAccountName, true)
+	Expect(err).ToNot(HaveOccurred())
+
+	_, _ = managedFw.AsKubeAdmin.CommonController.CreateRole("role-release-service-account", managedNamespace, map[string][]string{
+		"apiGroupsList": {""},
+		"roleResources": {"secrets"},
+		"roleVerbs":     {"get", "list", "watch"},
+	})
+
+	_, _ = managedFw.AsKubeAdmin.CommonController.CreateRoleBinding("role-release-service-account-binding", managedNamespace, "ServiceAccount", "release-service-account", managedNamespace, "Role", "role-release-service-account", "rbac.authorization.k8s.io")
+
+	defaultEcPolicySpec := ecp.EnterpriseContractPolicySpec{
+		Description: "Red Hat's enterprise requirements",
+		PublicKey:   fmt.Sprintf("k8s://%s/%s", managedNamespace, releasecommon.PublicSecretNameAuth),
+		Sources: []ecp.Source{{
+			Name:   "Default",
+			Policy: []string{releasecommon.EcPolicyLibPath, releasecommon.EcPolicyReleasePath},
+			Data:   []string{releasecommon.EcPolicyDataBundle, releasecommon.EcPolicyDataPath},
+		}},
+		Configuration: &ecp.EnterpriseContractPolicyConfiguration{
+			Exclude: []string{"step_image_registries", "tasks.required_tasks_found:prefetch-dependencies"},
+			Include: []string{"@slsa3"},
+		},
+	}
+
+	_, err := managedFw.AsKubeDeveloper.TektonController.CreateEnterpriseContractPolicy(rhaiECPName, managedNamespace, defaultEcPolicySpec)
+	Expect(err).NotTo(HaveOccurred())
+
+}
+
+func createRHAIReleasePlanAdmission(rhaiRPAName string, managedFw framework.Framework, devNamespace, managedNamespace, rhaiAppName, rhaiECPName, pathInRepoValue string) {
+	var err error
+
+	data, err := json.Marshal(map[string]interface{}{
+		"mapping": map[string]interface{}{
+			"components": []map[string]interface{}{
+				{
+					"name": rhaiComponentName,
+					"staged": map[string]interface{}{
+						"destination": "rhelai-1_DOT_1-for-rhel-9-x86_64-isos",
+						"version": "1.1",
+						"files": []map[string]interface{}{
```

Comment:
Did you try oras pulling the image we are using to make sure these are the 3 files in it (and no others)?

Reply:
From which URL to pull the image? I can't see any log message about the push result except the return code.

Reply:
The `containerImage` from the component. Check out https://github.com/hacbs-release/app-interface-deployments/blob/main/internal-services/catalog/pulp-push-disk-images-task.yaml#L173 it is what handles this

Reply:
Did you take a look at the task? If the PULLSPEC is a proper AI image, then when you oras pull it you will get a bunch of zipped files. We need to make sure each one is present in the `.files[]` key here

Reply:
I ever tried to pull it to my local desktop. I can see all the three files (install.iso.gz, disk.raw.gz and disk.qcow2.gz).  Do we need to add a step in the script to verify it?

Reply:
No. Sometimes there are more than 3 files. That is all I was asking about - that we have an entry for every file contained in the image and didn't miss any